#### Author: Diogo Viana

#### Data Scientist & Data Analyst

2024

Contact information
- https://github.com/DiogocViana\n
- https://www.linkedin.com/in/diogo-viana/

This dataset, acquired from Kaggle (https://www.kaggle.com/datasets/dillonmyrick/bike-store-sample-database/data), represents a sample dataset for a retail bike store sourced from sqlservertutorial.net. Within this notebook, I aim to showcase my proficiency in working with SQL datasets and querying relational databases using Python. Please note that this notebook is a work in progress, and I anticipate refining this introduction section in subsequent iterations

In [1]:
import pandas as pd
import sqlite3

In [5]:
#After downloading the relational database, import all its tables

brands = pd.read_csv('brands.csv')
categories = pd.read_csv('categories.csv')
customers = pd.read_csv('customers.csv')
order_items = pd.read_csv('order_items.csv')
orders = pd.read_csv('orders.csv')
products = pd.read_csv('products.csv')
staffs = pd.read_csv('staffs.csv')
stocks = pd.read_csv('stocks.csv')
stores = pd.read_csv('stores.csv')

In [6]:
#Create the database conection
connection = sqlite3.connect('bike_store.db')

# Insert data into database
brands.to_sql('brands', connection, if_exists='replace', index=False)
categories.to_sql('categories', connection, if_exists='replace', index=False)
customers.to_sql('customers', connection, if_exists='replace', index=False)
order_items.to_sql('order_items', connection, if_exists='replace', index=False)
orders.to_sql('orders', connection, if_exists='replace', index=False)
products.to_sql('products', connection, if_exists='replace', index=False)
staffs.to_sql('staffs', connection, if_exists='replace', index=False)
stocks.to_sql('stocks', connection, if_exists='replace', index=False)
stores.to_sql('stores', connection, if_exists='replace', index=False)

3

Overview of each table

In [7]:
#Brand Table
query = 'SELECT * FROM  brands'

df = pd.read_sql_query(query, connection)
df

brand_id    brand_name
0         1       Electra
1         2          Haro
2         3        Heller
3         4   Pure Cycles
4         5       Ritchey
5         6       Strider
6         7  Sun Bicycles
7         8         Surly
8         9          Trek

In [11]:
#Brand Table
query = 'SELECT COUNT (*) FROM  brands'

df = pd.read_sql_query(query, connection)
df

COUNT (*)
0          9

As we can see, there are 9 rows in the brand table.

In [8]:
#Category Table
query = 'SELECT * FROM  categories'

df = pd.read_sql_query(query, connection)
df

category_id        category_name
0            1    Children Bicycles
1            2     Comfort Bicycles
2            3    Cruisers Bicycles
3            4  Cyclocross Bicycles
4            5       Electric Bikes
5            6       Mountain Bikes
6            7           Road Bikes

In [12]:
#Brand Table
query = 'SELECT COUNT (*) FROM  categories'

df = pd.read_sql_query(query, connection)
df

COUNT (*)
0          7

As we can see, there are 7 rows in the categories table.

In [13]:
#Customers Table
query = 'SELECT * FROM  customers LIMIT 5'

df = pd.read_sql_query(query, connection)
df

customer_id  first_name last_name           phone                    email  \
0            1       Debra     Burks            None    debra.burks@yahoo.com   
1            2       Kasha      Todd            None     kasha.todd@yahoo.com   
2            3      Tameka    Fisher            None    tameka.fisher@aol.com   
3            4       Daryl    Spence            None     daryl.spence@aol.com   
4            5  Charolette      Rice  (916) 381-6003  charolette.rice@msn.com   

                  street           city state  zip_code  
0      9273 Thorne Ave.    Orchard Park    NY     14127  
1       910 Vine Street        Campbell    CA     95008  
2  769C Honey Creek St.   Redondo Beach    CA     90278  
3        988 Pearl Lane       Uniondale    NY     11553  
4         107 River Dr.      Sacramento    CA     95820

In [14]:
#Customers Table
query = 'SELECT COUNT (*) FROM  customers'

df = pd.read_sql_query(query, connection)
df

COUNT (*)
0       1445

The 'customer' table comprises 9 columns and 1445 rows. Let's examine the number of unique values in certain columns within this table.

In [22]:
#Customers Table
query = 'SELECT DISTINCT first_name FROM  customers'

df = pd.read_sql_query(query, connection)
df

first_name
0          Debra
1          Kasha
2         Tameka
3          Daryl
4     Charolette
...          ...
1260         Son
1261     Florrie
1262      Ernest
1263      Cassie
1264       Ester

[1265 rows x 1 columns]

In [23]:
#Customers Table
query = 'SELECT DISTINCT email FROM  customers'

df = pd.read_sql_query(query, connection)
df

email
0       debra.burks@yahoo.com
1        kasha.todd@yahoo.com
2       tameka.fisher@aol.com
3        daryl.spence@aol.com
4     charolette.rice@msn.com
...                       ...
1440  jamaal.morrison@msn.com
1441   cassie.cline@gmail.com
1442    lezlie.lamb@gmail.com
1443   ivette.estes@gmail.com
1444  ester.acevedo@gmail.com

[1445 rows x 1 columns]

In [24]:
#Customers Table
query = 'SELECT DISTINCT city FROM  customers'

df = pd.read_sql_query(query, connection)
df

city
0       Orchard Park
1           Campbell
2      Redondo Beach
3          Uniondale
4         Sacramento
..               ...
190    Copperas Cove
191  Oakland Gardens
192        Tonawanda
193     Far Rockaway
194         Westbury

[195 rows x 1 columns]

Upon reviewing the outcomes of the three queries mentioned earlier, it becomes apparent that there are 180 individuals with identical names. Therefore, considering an alternative column or merging it with another is necessary for a more in-depth investigation. Conversely, the email column appears to be unique for each customer. Furthermore, a noteworthy observation is that the city information, gleaned from the above queries, reveals 195 distinct cities, which may prove crucial for future analyses.

In [25]:
#Order_items Table
query = 'SELECT * FROM order_items LIMIT 5'

df = pd.read_sql_query(query, connection)
df

order_id  item_id  product_id  quantity  list_price  discount
0         1        1          20         1      599.99      0.20
1         1        2           8         2     1799.99      0.07
2         1        3          10         2     1549.00      0.05
3         1        4          16         2      599.99      0.05
4         1        5           4         1     2899.99      0.20

In [26]:
#Order_items Table
query = 'SELECT COUNT (*) FROM  order_items'

df = pd.read_sql_query(query, connection)
df

COUNT (*)
0       4722

In [27]:
#Customers Table
query = 'SELECT DISTINCT discount FROM order_items'

df = pd.read_sql_query(query, connection)
df

discount
0      0.20
1      0.07
2      0.05
3      0.10

The 'order_items' table encompasses 6 columns and consists of 4722 rows. Significantly, it is notable that the provided discounts exhibit variation, ranging between 5%, 7%, 10%, and 20%.

To be continued...